rfcs/0151-nixos-port-alloc.md

```diff
@@ -0,0 +1,135 @@
+---
+feature: nixos-port-alloc
+start-date: 2023-06-10
+author: lucasew
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+A port allocator for NixOS services.
+
+# Motivation
+[motivation]: #motivation
+
+Sometimes people don't care about which port a service is running, only that it
+should be written somewhere so a service such as nginx can find it.
+
+# Detailed design
+[design]: #detailed-design
+
+The less likely used ports of the port space in a system are the higher ones,
+and the highest one is 65535, so a NixOS module could keep track of which services
+need a port and the service modules need only to reference that port in their
+configurations.
+
+This module exposes the options under `networking.ports`. A service module can
+request a port by defining `networking.ports.service.enable = true` and get the
+allocated port by referring to `networking.ports.service.port`. The service doesn't
```

Comment:
```suggestion
request a port by defining `networking.ports.${service}.enable = true` and get the
allocated port by referring to `networking.ports.${service}.port`. The service doesn't
```

On first read it wasn't clear to me that `service` was a placeholder. It looks similar to `services.foo` or `systemd.services.foo` where it is a literal. I think it would be good to make it more clear.

Alternative suggestions are `networking.ports.foo`, `networking.ports.someservice` , `networking.ports.${service-name}`, `networking.ports.*`, ``networking.ports.SERVICE`. I think either of these would be more obviously a placeholder.